src/Adapter/MSTest.TestAdapter/Execution/TestMethodRunner.cs

```diff
@@ -254,6 +249,44 @@ internal UnitTestResult[] RunTestMethod()
         return results.ToUnitTestResults();
     }
 
+    private bool TryExecuteITestDataSource(List<TestResult> results)
+    {
+        if (_test.DataType != DynamicDataType.ITestDataSource)
+        {
+            return false;
+        }
+
+        UTF.ITestDataSource? dataSource;
+        object?[]? data;
+        if (_test.SerializedData?.Length == 3)
+        {
+            if (!Enum.TryParse(_test.SerializedData[0], out TestDataSourceUnfoldingStrategy _)
+                || !int.TryParse(_test.SerializedData[1], out int dataSourceIndex)
+                || !int.TryParse(_test.SerializedData[2], out int dataIndex))
```

Comment:
Consider using the `IFormatProvider` overload and pass `CultureInfo.InvariantCulture`.

Reply:
I'm surprised there was no CA warning here. I need to verify the existing rules.